File: client/src/types/governance.ts
// src/types/governance.ts

export type AutonomyLevel =
  | 'Nenhum'
  | 'Apenas Ferramentas Seguras'
  | 'Semi-Autônomo'
  | 'Totalmente Autônomo'

export interface ApprovalItem {
  id: string
  agentName: string
  action: string
  context: string
  status: 'pending' | 'approved' | 'rejected'
  createdAt: string
}

export interface HistoryItem {
  id: string
  agentName: string
  action: string
  status: 'approved' | 'rejected'
  timestamp: string
  actor: string
  reason?: string
}

export interface AuditLogActor {
  type: 'user' | 'agent'
  id: string
  name: string
}

export interface AuditLog {
  id: string
  timestamp: string
  actor: AuditLogActor
  action: string
  details: Record<string, unknown>
  ipAddress?: string
}

File: server/app/models/governance.py
from pydantic import BaseModel
from typing import Any, Dict, Optional, Literal
from datetime import datetime

class GovernanceSettings(BaseModel):
    defaultAutonomyLevel: Literal['auto', 'ask']

class ApprovalItem(BaseModel):
    id: str
    agentName: str
    action: str
    details: Dict[str, Any]
    status: Literal['pending', 'approved', 'rejected'] = 'pending'
    timestamp: datetime

class HistoryItem(BaseModel):
    id: str
    agentName: str
    action: str
    status: Literal['approved', 'rejected']
    timestamp: datetime
    actor: str
    reason: Optional[str] = None

class RejectPayload(BaseModel):
    reason: str
